### ISSUE TITLE
Layer normalization on Nested Tensor ragged dimension fails when `lengths is not None`

### ISSUE THREAD

**Issue by jananisriram** [pytorch/pytorch#131967]
### 🐛 Describe the bug

Performing layer normalization along the ragged dimension on a nested tensor with holes (`lengths is not None`) is currently incorrect but does not throw an error:
```
import torch

device, dtype = "cpu", torch.float32
lengths = torch.randint(5, 10, (20,), device=device)
offsets = torch.zeros((21,), device=device, dtype=torch.int)
torch.cumsum(lengths, dim=0, out=offsets[1:])
values = torch.randn(
    (offsets[-1].item(), 10, 30),
    device=device,
    dtype=dtype,
)

nt_with_holes = torch.nested.nested_tensor_from_jagged(
    values,
    offsets,
    lengths=offsets.diff() - 2,  # arbitrary subtraction to create holes
)

out = torch.nn.functional.layer_norm(
    nt_with_holes, normalized_shape=(-1, 10, 30),
)
```

### Versions

fbcode H100 7/26/24
